my_compassion/models/res_user.py

```diff
@@ -31,3 +34,176 @@ def reset_password(self, login):
+        )
+
+        if partners:
+            # Select the best existing partner
+            selected_partner = partners.sorted(
+                key=lambda partner: (
+                    partner.number_sponsorships,
+                    partner.write_date or partner.create_date,
+                ),
+                reverse=True,
+            )[0]
+            user.write({"partner_id": selected_partner.id})
+            _logger.info(
+                "User %s linked to existing partner %s asynchronously.",
+                user.login,
+                selected_partner.name,
+            )
+            selected_partner.signup_type = old_partner.signup_type
+            selected_partner.signup_token = old_partner.signup_token
+            selected_partner.signup_expiration = old_partner.signup_expiration
+            selected_partner.signup_url = old_partner.signup_url
+            old_partner.unlink()
+            _logger.info("Partner deleted.")
+
+            _logger.info(
+                "User %s linked to existing partner %s asynchronously.",
+                user.login,
+                selected_partner.name,
+            )
+        else:
+            # No better partner found
+            _logger.info("No better partner found for user %s.", user.login)
+
+        self.flush()
+
+    @api.model
+    def signup(self, values, token=None):
+        """Override signup to avoid creating unnecessary partners."""
+        values.pop("privacy_policy", None)
+
+        if token:
+            # Existing logic for token-based signup
+            partner = self.env["res.partner"]._signup_retrieve_partner(
+                token, check_validity=True, raise_exception=True
+            )
+            partner.write(
+                {
+                    "signup_token": False,
+                    "signup_type": False,
+                    "signup_expiration": False,
+                }
+            )
+
+            partner_user = partner.user_ids and partner.user_ids[0] or False
+
+            if partner_user:
+                values.pop("login", None)
+                values.pop("name", None)
+                partner_user.write(values)
+                if not partner_user.login_date:
+                    partner_user._notify_inviter()
+                return (self.env.cr.dbname, partner_user.login, values.get("password"))
+            else:
+                values.update(
+                    {
+                        "name": partner.name,
+                        "partner_id": partner.id,
+                        "email": values.get("email") or values.get("login"),
+                    }
+                )
+                if partner.company_id:
+                    values["company_id"] = partner.company_id.id
+                    values["company_ids"] = [(6, 0, [partner.company_id.id])]
+                partner_user = self._signup_create_user(values)
+                partner_user._notify_inviter()
```

Comment:
Can't we use the super() call in this case? It's a better practice rather than re-writing existing code.